src/ui/Services/CourseMapper.cs

```diff
@@ -95,7 +100,7 @@ public SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(ApiClie
                 ApplicationsAcceptedFrom = ucasCourseData.Schools.Select(x => {
                         DateTime parsed;                    
                         return DateTime.TryParse(x.ApplicationsAcceptedFrom, out parsed) ? (DateTime?) parsed : null;
-                    }).Where(x => x != null)
+                    }).Where(x => x != null && x.HasValue)
```

Comment:
I think the extra clause is redundant.

ref https://stackoverflow.com/questions/17783929/c-sharp-hasvalue-vs-null/17784002#17784002

minor